python/ray/tune/trainable/trainable.py

```diff
@@ -420,6 +422,14 @@ def train(self):
 
         return result
 
+    @property
+    def _cloud_checkpoint_num_retries(self) -> int:
+        return int(os.getenv("TUNE_CHECKPOINT_CLOUD_RETRY_NUM", "3"))
+
+    @property
+    def _cloud_checkpoint_sleep_time(self) -> int:
+        return float(os.getenv("TUNE_CHECKPOINT_CLOUD_RETRY_WAIT_TIME_S", "1"))
+
```

Comment:
Is there a reason for us to not just store it once on initialization?

Reply:
The value will be persisted during serialization then, which doesn't make sense. I feel like enviroment variables, like the name suggests, are a part of the enviroment in which the object runs and not the state of the object itself, and therefore should not be stored as attributes

Reply:
But we only serialize trainable classes, not objects, right?

Reply:
E.g. this works:

```
import ray
import os


@ray.remote
class Actor:
    def __init__(self):
        self.my_var = os.environ.get("FOO")

    def get(self):
        return self.my_var


os.environ["FOO"] = "bar"

actor = Actor.remote()
print(ray.get(actor.get.remote()))
```

as does this:

```
import ray
import os


@ray.remote
class Actor:
    def __init__(self):
        self.my_var = os.environ.get("FOO")

    def get(self):
        return self.my_var


actor = Actor.options(runtime_env={"env_vars": {"FOO": "baz"}}).remote()
print(ray.get(actor.get.remote()))
```

Reply:
We do serialize parts of the Trainable object using `Trainable.get_state` but not the entire object, so I guess it's fine to save them as attributes